File: src/HandWinnerComputer.cs
class HandWinnerComputer
{
    public static Card calculateWinner(List<Card> cards, Card.SUIT briscolaSuit)
    {
        if (!cards.Any())
            return null;
        if (1 == cards.Length)
            return cards[0];

        Card winner = cards[0];
        for (int i = 1; i < cards.Length; i++) {
            Card currCard = cards[i];
            winner = Card.getWinner(winner, currCard, briscolaSuit);
        }
        return winner;
    }
}

File: briscola/lib/hand_winner_computer.rb
module HandWinnerComputer
  def self.calculate_winner(cards, briscola_suit)
    return nil if cards.empty?
    return cards.first if cards.length == 1
    cards[1..].reduce(cards.first) { |winner, card| Card.get_winner(winner, card) }
  end
end
